### PR TITLE
Fixup clippy lints

### PR DESCRIPTION
**Pull request by PolyMeilex**

### CODE REVIEW

File: src/wayland/shm/pool.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#![forbid(unsafe_op_in_unsafe_fn)]
```
Comment (PolyMeilex) [inline src/wayland/shm/pool.rs:1]: forbid was causing a conflict somewhere inside of thread_local macro, so I changed it to deny

File: src/wayland/shm/pool.rs [hunk 2]
```diff
@@ -44,7 +44,7 @@ static DROP_THIS: Lazy<SyncSender<InnerPool>> = Lazy::new(|| {
     tx
 });
 
-thread_local!(static SIGBUS_GUARD: Cell<(*const MemMap, bool)> = Cell::new((ptr::null_mut(), false)));
+thread_local!(static SIGBUS_GUARD: Cell<(*const MemMap, bool)> = const { Cell::new((ptr::null_mut(), false)) });
```
Comment (i509VCB) [inline src/wayland/shm/pool.rs:47]: This changes the MSRV?

Comment (PolyMeilex) [reply src/wayland/shm/pool.rs:47]: Not 100% sure, but I don't think so: this is 3 years old https://github.com/dtolnay/rust/commit/c6eea222a9cafeeeedfa1767232328710b350447

**Review by Drakulix — APPROVED** [review]

File: anvil/src/input_handler.rs [hunk 3]
```diff
@@ -1306,6 +1306,7 @@ impl AnvilState<UdevData> {
 }
 
 /// Possible results of a keyboard action
+#[allow(dead_code)] // some of these are only read if udev is enabled
```
Comment (chrisduerr) [inline anvil/src/input_handler.rs:1309]: Are they always written to? Why not feature gate the variants behind the backend?

Comment (Drakulix) [reply anvil/src/input_handler.rs:1309]: Yes the corresponding key binding will still generate the KeyAction. Of course we could feature-gate all of that as well, but I am not sure that is cleaner than this short line.

**Comment by PolyMeilex** [comment]
Your commits look fine to me, so I'm just going to merge this.

**Pull request merged**